### PR TITLE
Ship database with attack mitigations

### LINKED ISSUE #629: Ship island mongodb with attack_mitigations already present
Parsing mitre CTI to extract and save `attack_mitigations` on the database takes some time (~4 seconds to me). This needlessly prolongs setup of monkey island. We should ship island binaries with `attack_mitigations` already present in the database and save some time for the user.

# Tasks

- [ ] Write a Python utility to generate mongo dump/export for Attack Mitigations. (See [this](https://github.com/guardicore/monkey/issues/629#issuecomment-919056350))  (0.25d)  @mssalvatore 
    - [x] Clean/clear/setup database
    - [x] Read attack data submodule and populate mongo database 
    - [x] Export/dump mongo database
    - [x] Remove unused import code from Monkey Island
    - [x] Add metadata including git commit and origin url in `attack_metadata.json`
    - [x] Document this in ??developer documentation??
- [x] Remove attack-data submodule and store the mongo dump in `monkey_island/cc/setup/mongo/` (0d) @ilija-lazoroski 
  - [x] Modify pyinstaller hooks to include mongo dump file  
  - [x] Fix monkey_island.spec
- [ ]  Replace the code in Monkey Island that populates the attack section with code that imports a dump and test all installers. (0.25d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do? 

Fixes #629 . 

Add any further explanations here. 

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

**Review by mssalvatore — APPROVED** [review]

File: docs/content/development/attack_mitigations.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+draft: true
+weight: 10
+---
+
+{{% notice info %}}
+Check out [the documentation for the MITRE ATT&CK techniques as well]({{< ref "/reports/mitre" >}}).
+{{% /notice %}}
+
+## Summary
+
```
Comment (VakarisZ) [inline docs/content/development/attack_mitigations.md:13]: Add a section for general public explaining what "Attack mitigations" means. I'd do something like:
"Attack Mitigations are presented in ATT&CK report. They appear next to attack descriptions to suggest steps that can be taken to reduce the risk of that particular attack technique happening in the network. They also provide links for further reading about the mitigation on ATT&CK's official website."

File: docs/content/development/attack_mitigations.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+## Updating the MITRE ATT&CK mitigations data
+1. Clone the [MITRE Cyber Threat Intelligence
+   Repository](https://github.com/mitre/cti) or the [Guardicore
+   fork](https://github.com/guardicore/cti)
+   ```
+   $ CTI_REPO=$PWD/cti
+   $ git clone <REPO> $CTI_REPO
+   ```
+2. Start a mongodb v4.2 server
```
Comment (VakarisZ) [inline docs/content/development/attack_mitigations.md:29]: Can I just start monkey island?

File: deployment_scripts/dump_attack_mitigations/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+charset-normalizer==2.0.6
+idna==3.2
+mongoengine==0.23.1
+pymongo==3.12.0
+pytz==2021.1
+requests==2.26.0
+simplejson==3.17.5
+six==1.16.0
+stix2==3.0.1
+stix2-patterns==1.3.2
```
Comment (VakarisZ) [inline deployment_scripts/dump_attack_mitigations/requirements.txt:12]: Does this mean that we no longer need `stix2` in island?

**Review by VakarisZ — APPROVED** [review]
Approved, but make sure to review my comments/questions!

File: docs/content/development/attack_mitigations.md (hunk 2 above)
Comment (ilija-lazoroski) [reply docs/content/development/attack_mitigations.md:29]: Yep, but this is separate utility. Nothing to do with starting the Island. Maybe a note saying something like "Starting Monkey Island will also start a mongo server" ?

File: deployment_scripts/dump_attack_mitigations/requirements.txt (hunk 3 above)
Comment (mssalvatore) [reply deployment_scripts/dump_attack_mitigations/requirements.txt:12]: It does.

**Pull request merged**